### ISSUE TITLE
Gametype selection

### ISSUE THREAD

**Issue by Dune-jr** [teeworlds/teeworlds#2705]
So there's been a lot of fuss about being unable to filter out specific gametypes since a lot of 0.6 servers have bridged into 0.7.
The issue is fixed thanks to @Robyt3's https://github.com/teeworlds/teeworlds/pull/2622

I have no problem with the way it is displayed in the server browser, but the selection looks tedious. 
Would you want to have a popout screen similar to country selection that automatically lists (all) server gametypes?
Ugly mockup where stuff isn't aligned:
![image](https://user-images.githubusercontent.com/355114/92176505-e1173900-ee3e-11ea-8bb8-fadd54c2d715.png)

The stuff in parenthesis could refer to the amount of servers or players, whichever.

**Comment by Robyt3** [comment]
I see how that would make it easier to use because you can see the exact games type names.

I assume it should always show the pure game types and then list all game types of servers in the current server browser tab, as well as existing game type filters even when no server currently has that type.

Is the checkbox supposed to be tri-state (nothing, [green]include[/green], [red]exclude[/red]) or should there be two checkboxes for include and exclude?

The default would be to include everything by not filtering, so the initial selection should just be empty. Selecting and deselecting everything would have basically the same effect as the existing "Clear" button, which could also be added to this popup.

**Comment by Learath2** [comment]
I agree that you should list vanilla types at the top, after that could sort by either total player count or total server count (I'd take player count, there are already people salty about the amount of servers we host).

A tristate checkbox sounds good, but might be too confusing to the common computer user. Though I'm not sure how the UX would be without it.

**Comment by Dune-jr** [comment]
I don't see the difference between nothing and include. It's a filter, it's binary: show the server or don't. 

Sorting is a separated thing.

**Comment by oy** [comment]
It's about making the user handling as simple as possible.
With a tristate checkbox:
- if just want to see `ctf`and `dm`you set these 2 to green
- if you want to see everything but `abc`and `bcd` you set these 2 to red
and you don't have to select/unselect anything else.
With a binary checkbox you'd have to do more selecting/unselecting.

**Comment by TsFreddie** [comment]
> With a binary checkbox you'd have to do more selecting/unselecting.

A "reverse selection" button is pretty common UX element, and you don't have to think about what does mixing green and red selection means (and it probably means nothing).
